packages/next/src/server/after/after-context.ts

```diff
@@ -83,7 +94,9 @@ export class AfterContext {
     //   await x()
     const wrappedCallback = bindSnapshot(async () => {
       try {
-        await callback()
+        await afterTaskAsyncStorage.run({ rootTaskSpawnPhase }, () =>
```

Comment:
I don't see why this extra storage is needed here, what does it preserve that the underlying `workUnitStore.phase` does not have?

Reply:
`workUnitStore.phase` is mutable, so by the time this callback runs, `workUnitStore.phase` will have changed from `action` or `render` to `after`, so we can't use it to determine whether a `headers()` call is legal in the current context